### DISCUSSION TITLE
Ligatures at Prompt vs in Editors

### DISCUSSION THREAD

**Discussion by Ducky6944** [wezterm/wezterm#1248] [Q&A] [answered]
I am still going through the documentation, but for some reason a subset of ligatures seem to work at a prompt, but all ligatures work in NVim, Nano, etc. Any Ideas? 

```LUA
-- W E Z T E R M

-- TODOS:
-- Learn How to name window, but not tab
-- Remove Title Bar
-- Live Reload
-- Change Colorschemes with Wal
local wezterm = require 'wezterm';
local home = wezterm.home_dir;
wezterm.add_to_config_reload_watch_list(home.."/.cache/wal/wezterm-wal.toml")
return {
  -- Wal Colorscheme NOTE: you could definte multiple colorschemes in this file 
  color_scheme_dirs = {home.."/.cache/wal"},
  hide_tab_bar_if_only_one_tab = true,
  color_scheme = "wezterm-wal",
  font = wezterm.font_with_fallback({"Iosevka Nerd Font","Apple Color Emoji"}),
  font_size = 18.0,
  freetype_interpreter_version = 40,
  font_shaper = "Harfbuzz",
  send_composed_key_when_left_alt_is_pressed=false,
  send_composed_key_when_right_alt_is_pressed=true,
  native_macos_fullscreen_mode = true,
  allow_square_glyphs_to_overflow_width = "Never",
  adjust_window_size_when_changing_font_size = false,
  use_ime = false,
  window_decorations = "RESIZE",
  window_padding = {
    left = 12,
    -- This will become the scrollbar width if you have enabled the scrollbar!
    right = 12,
    top = 12,
    bottom = 12,
  },
  leader = { key="a", mods="CTRL", timeout_milliseconds=1000 },
  -- keys = {
    -- Figure out how to move left and right in terminal with cmd and opt
  -- }
  -- wezterm.add_to_config_reload_watch_list(home.."/.cache/wal"),
  -- Test
  	font_rules={
		{
			-- Italic
			italic=true,
			font = wezterm.font_with_fallback( {
				{family="Iosevka Nerd Font", italic=true},
				"Inconsolata Nerd Font Mono",
				"Noto Color Emoji",
			}),
		},
		{
			-- Bold
			intensity="Bold", italic=false,
			font = wezterm.font_with_fallback( {
				{family="Iosevka Nerd Font", weight="Bold"},
				"Inconsolata Nerd Font Mono",
				"Noto Color Emoji",
			}),
		},
		{
			-- Bold Italic
			intensity="Bold", italic=true,
			font = wezterm.font_with_fallback( {
				{family="Iosevka Nerd Font", weight="Bold", italic=true},
				"Inconsolata Nerd Font Mono",
				"Noto Color Emoji",
			}),
		},
	},

	-- Font options
	harfbuzz_features = {
		"dlig", -- Ligatures
		-- "ss01", -- Single-story a
		-- "ss02", -- Single-story g
		-- "ss03", -- Simplified f
		-- "ss04", -- Simplified i
		-- "ss05", -- Simplified l
		-- "ss06", -- Simplified
		-- "ss07", -- Simplified italic diagonals (kwxyz)
		-- "ss08", -- No-serif L and Z
		-- "ss09", -- Simplified 6 and 9
		-- "ss10", -- Dotted 0
		-- "ss11", -- Simplified 1
		-- "ss12", -- Simplified @
    -- "calt=1",
    -- "clig=1"
	},

	-- Font rendering
	freetype_render_target = "Light",

	-- Cursor style
	default_cursor_style = "BlinkingBar",

	-- Hide tab bar when there is only a single tab
	hide_tab_bar_if_only_one_tab = true,

	-- Transparency
	window_background_opacity = 1.0,


}
```
In other terminals such as kitty they work at prompt and in editors. 
<img width="327" alt="Screen Shot 2021-10-22 at 2 10 26 PM" src="https://user-images.githubusercontent.com/12100960/138503466-e0b43bbd-f673-4411-9fb9-f8b91375cd76.png">

_WezTerm in Nvim_

<img width="293" alt="Screen Shot 2021-10-22 at 2 09 22 PM" src="https://user-images.githubusercontent.com/12100960/138503482-554a91cc-fdf2-4f2e-8da6-e064cf2f8489.png">

_WezTerm at Prompt_

I also just noticed that ligatures aren't completely working until after some text has been typed at the prompt.  For Example in the screen shot below is all just tildas`~~~ ~~~~~~`.

<img width="194" alt="Screen Shot 2021-10-22 at 2 22 34 PM" src="https://user-images.githubusercontent.com/12100960/138504543-4e34807a-e110-4117-bcc8-eac3e48ba69c.png">

_WezTerm at prompt_ 

I've tried various configurations of dlig clig and other options that looked like they may be relevant, but from the documentation it seems harfbuzz should apply most of the applicable options by default. I've also tried several fonts, with similar results.

**Comment by bew** [comment, marked as answer]
This is because the ligatures are split by a color change!

For example, for the `>=` ligature, where `>` and `=` are colored differently by your shell, the terminal cannot join them.. How do you think it should look like?

**Reply by Ducky6944** [reply]
That's interesting. I assume because the syntax highlighter is adding escape codes of something between the actual characters right? Using bash where I have no config works as expected. I guess I should either turn off syntax highlighting or just deal with it since ligatures aren't really needed for me at the prompt anyway. Thanks for the answer!
